pkg/module/template.go

```diff
@@ -38,13 +39,25 @@ spec:
 {{yaml .Descriptor | printf "%s" | indent 4}}
 `
 
-func Template(archive *ctf.ComponentArchive, channel string, data []byte) ([]byte, error) {
-	d, err := remoteDescriptor(archive)
+	OCIRegistryCredLabel = "oci-registry-cred"
+	credSecretLabel      = "operator.kyma-project.io/oci-registry-cred"
+)
+
+func Template(archive *ctf.ComponentArchive, channel string, data []byte, enableRegistryCred bool) ([]byte, error) {
+	descriptor, err := remoteDescriptor(archive)
 	if err != nil {
 		return nil, err
 	}
-
-	ref, err := oci.ParseRef(d.Name)
+	if enableRegistryCred {
+		for i := range descriptor.Resources {
+			resource := &descriptor.Resources[i]
+			resource.SetLabels([]v2.Label{{
+				Name:  OCIRegistryCredLabel,
+				Value: json.RawMessage(fmt.Sprintf(`{"%s": "%s"}`, credSecretLabel, descriptor.Name)),
```

Comment:
To Reviewer:
The reason why I choose the way not let users to define the label by themselves is a) simplify the cli command, so that users don't have to write labels in the command, b) avoid label validation. c) provide a standard label which can be used by module teams to provide it in the credential secret.

Of cause if user still have the use case to use customized label, they can simply manually overwrite the label in the generated module template, but not recommend.

Reply:
I believe your main intention is good and I support having a fixed label for the registry credential. However, I believe that also fixing the registry value is not ideal. Especially because that means that every module template will need its own secret which is pretty counter-intuitive and hard to maintain. I suggest to leave the value open for users to fill and only have the key defined. That way multiple modules will be able to use the same registry credential if desired. WDYT?

Reply:
Yes, you are right, to use a shared secret is not supported by this way. But just to clarify, I think for security concern we should encourage teams use their own repository credential if necessary compare to share the credential, or? 

But to leave the value customizable is fine to me, I'll change the flag to something like `--registry-cred-selector-value`, is it fine?

Reply:
As discussed, I made the change to accept the label selectors as flag input.